### PR TITLE
Feature post

### PR DESCRIPTION
**Pull request by ANGELLOPARR**
Very basic comment form on my front page. Only includes comment body for now but should soon include other fields after I get Datastore. Makes use of doPost() and doGet() on load of the page to display comments.

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -31,9 +31,6 @@
   @Override
   public void init() {
     comments = new ArrayList<String>();
```
Comment (pavlik-y) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:33]: Alternatively, if you only need to instantiate comments object, but not populate it you can do so in line 29 where you declare it:
private ArrayList<String> comments = new ArrayList<>();

Either way is fine.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -45,4 +42,24 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     response.setContentType("application/json");
     response.getWriter().println(json);
   }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get body of comment
+    String newComment = getParameter(request, "comment-body", "");
```
Comment (pavlik-y) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:49]: Not part of the assignment, but still a good thought exercise. What will happen if the user clears comment-body text area and hits submit?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (ANGELLOPARR) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:33]: Did not think of that, thank you!

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 2 above)
Comment (ANGELLOPARR) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:49]: It should put the empty comment in the array according to the code, but that should result in a warning or message to the user that it must be filled out! Now how to fix that...

Comment (pavlik-y) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:49]: It usually depends on the application, but in this particular case I think posting empty comment is invalid scenario. I'd say it shouldn't even reach the server. Ideally there should be some code on the client that only posts when comment is non-empty. I'm pretty sure there is a solution for that (typical problems should have standard solutions), but I don't know the solution, will need to research it a bit.

Comment (ANGELLOPARR) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:49]: Gotcha, I am currently looking into a javascript function that verifies required elements have some value before reaching the backend.

Comment (pavlik-y) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:49]: https://developer.mozilla.org/en-US/docs/Learn/Forms/Form_validation

Comment (ANGELLOPARR) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:49]: Oh this is actually really handy. HTML5 has built-in things to deal with this.

**Pull request merged**